src/measure_theory/ae_measurable_sequence.lean

```diff
@@ -0,0 +1,149 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+
+import measure_theory.measure_space
+
+/-!
+# Sequence of measurable functions associated to a sequence of a.e.-measurable functions
+
+We define here tools to prove statements about limits (infi, supr...) of sequences of
+`ae_measurable` functions.
+Given a sequence of a.e.-measurable functions `f : ι → α → β` with hypothesis
+`hf : ∀ i, ae_measurable (f i) μ`, and a pointwise property `p : α → (ι → β) → Prop` such that we
+have `hp : ∀ᵐ x ∂μ, p x (λ n, f n x)`, we define a sequence of measurable functions `ae_seq hf p`
+and a measurable set `ae_seq_set hf p`, such that
+* `μ (ae_seq_set hf p)ᶜ = 0`
+* `x ∈ ae_seq_set hf p → ∀ i : ι, ae_seq hf hp i x = f i x`
+* `x ∈ ae_seq_set hf p → p x (λ n, f n x)`
+-/
+
+open measure_theory
+open_locale classical
+
+variables {α β γ ι : Type*} [measurable_space α] [measurable_space β]
+  {f : ι → α → β} {μ : measure α} {p : α → (ι → β) → Prop}
+
+/-- If we have the additional hypothesis `∀ᵐ x ∂μ, p x (λ n, f n x)`, this is a measurable set
+whose complement has measure 0 such that for all `x ∈ ae_seq_set`, `f i x` is equal to
+`(hf i).mk (f i) x` for all `i` and we have the pointwise property `p x (λ n, f n x)`. -/
+def ae_seq_set (hf : ∀ i, ae_measurable (f i) μ) (p : α → (ι → β) → Prop) : set α :=
+(to_measurable μ {x | (∀ i, f i x = (hf i).mk (f i) x) ∧ p x (λ n, f n x)}ᶜ)ᶜ
+
+/-- A sequence of measurable functions that are equal to `f` and verify property `p` on the
+measurable set `ae_seq_set hf p`. -/
+noncomputable
+def ae_seq [hβ : nonempty β] (hf : ∀ i, ae_measurable (f i) μ) (p : α → (ι → β) → Prop) :
```

Comment:
You don't need the assumption that `β` is nonempty here: in the function definition, you have `i` and `x`, so you can prove that `β` is nonempty by using `f i x`. I don't know if it helps to simplify some proofs down the road.

Reply:
It helped! There is still an empty case in is_lub/is_glb, but the others are gone.